extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -118,7 +118,8 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
                     throw new RuntimeException(e);
                 }
             }
-            Handler<RoutingContext> handler = recorder.handler(cached.cachedDirectory, swaggerUiConfig.path);
+            Handler<RoutingContext> handler = recorder.handler(cached.cachedDirectory,
+                    httpRootPathBuildItem.adjustPath(swaggerUiConfig.path));
             routes.produce(new RouteBuildItem(swaggerUiConfig.path, handler));
             routes.produce(new RouteBuildItem(swaggerUiConfig.path + "/*", handler));
```

Comment:
The routes don't need the adjusted path?

Reply:
It is handled automatically, as the underlying router is mounted at the http root.